builtin/mainmenu/tab_online.lua

```diff
@@ -229,6 +233,17 @@ local function main_button_handler(tabview, fields, name, tabdata)
 		return true
 	end
 
+	if fields.btn_add_favorite then
+		core.add_favorite({
+			name        = gamedata.servername,
+			address     = gamedata.address,
+			port        = gamedata.port,
+			description = gamedata.description
```

Comment:
This is causing a crash. `gamedata` doesn't seem to hold this information. It is in `serverlist[tabdata.fav_selected]` (if `tabdata.fav_selected` is not nil or 0).

Reply:
There's a `gamedata.serverdescription`.

Reply:
It's empty at that point. Also, I mean all four fields, but I didn't want to comment on all four lines :)

Have you tried the patch? It crashes for me due to this data being empty.